src/gen_alpha_poc/quaternion.h

```diff
@@ -308,32 +308,24 @@ Quaternion rotation_matrix_to_quaternion(const RotationMatrix& rotation_matrix)
 }
 
 /// Returns the B derivative matrix given for Euler parameters, i.e. unit quaternions
-inline Kokkos::View<double**> BMatrixForQuaternions(const Quaternion& quaternion) {
```

Comment:
I'm not clear on why you converted from return-value to in-out argument. Is it related to converting the View template parameter from `double**` to `double[3][4]` ? It seems like you could have done that either way.

Reply:
It could be done either way.  This was mostly a style choice to separate memory allocation from math.  The material change is changing the input argument from a `Quaternion` to a `Kokkos::View<const double[4]>` to ensure host/device memory consistency.  Since I was already changing the function signature, I went ahead and changed the output style, too.